DigitalLearningSolutions.Web/Helpers/CustomClaimHelper.cs

```diff
@@ -15,6 +15,11 @@ public static int GetUserIdKnownNotNull(this ClaimsPrincipal user)
             return user.GetCustomClaimAsRequiredInt(CustomClaimTypes.UserId);
         }
 
+        public static bool IsMissingUserId(this ClaimsPrincipal user)
+        {
+            return user.GetUserPrimaryEmail() != null && user.GetUserId() == null;

```

Comment:
I think it'd be more straightforward to do `user.Identity.IsAuthenticated`.

Reply:
Fair